### DISCUSSION TITLE
How to identify PDU controllable outlets?

### DISCUSSION THREAD

**Discussion by llamafilm** [netbox-community/netbox#14971] [Help Wanted!]
I have some PDUs where only _some_ outlets are controllable, for example [Middle Atlantic RLNK-1615V](https://res.cloudinary.com/avd/image/upload/v133509816/Resources/Middle%20Atlantic/Power/Specifications/96_01238_RLSelect_Vert.pdf).
I need to track which outlets are controllable and which are not, because I'm building a PDU rebooter service to kick failed devices.

If I could add a custom boolean field to the Power Outlet in the _Device Type_ that would address this nicely.  But that is not possible, see #12423.

I can add a keyword *Controllable* in the outlet name and use that as a filter in my rebooter service.  But I wonder if it would be worthwhile to add a new field in the database for this.

**Comment by candlerb** [comment]
You can add a custom field to Power Outlet, and you can mark it as "UI Visible: If Set". Then if you don't set it, then it won't clutter up the UI for other power outlets.

A Device Type has Power Outlet Templates. I'm not sure if custom fields on a Power Outlet Template are copied to the Power Outlet that is created when a device is instantiated (and if not, perhaps they should be).

But at worst, you'd just do a bulk edit of the Power Outlets and set the custom fields as required.

**Reply by llamafilm** [reply]
You can't add custom fields on power outlet templates.  (Or any other type of template, I think)

**Reply by candlerb** [reply]
Yes you're right. That would be a nice feature I think, and in principle ought to be fairly easy to implement (you'd just copy custom_field_data from the template to the object when instantiating it). The main problem I see would be ensuring in the UI that model pairs like [Power Outlet, Power Outlet Template] share the same custom field definitions.

But for now, you can add a custom field to Power Outlet, and then set the value of these custom fields manually when instantiating a PDU - or use a Custom Script to set them appropriately.
